cmd/juju/action/showoutput.go

```diff
@@ -130,7 +160,39 @@ func (c *showOutputCommand) Run(ctx *cmd.Context) error {
 		wait = time.NewTimer(waitDur)
 	}
 
+	actionDone := make(chan struct{})
+	var logsWatcher watcher.StringsWatcher
+	haveLogs := false
+
+	shouldWatch := waitDur.Nanoseconds() >= 0
+	if shouldWatch {
+		result, err := fetchResult(api, c.requestedId)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		shouldWatch = result.Status != params.ActionCompleted
```

Comment:
Should this only watch on Status == ActionPending?

Reply:
yes, yes it should